rir/src/compiler/test/PirCheck.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "R/Symbols.h"
+#include "common.h"
+#include <list>
+
+namespace rir {
+
+#define LIST_OF_PIR_CHECKS(V)                                                  \
```

Comment:
I mean, this trick is nice and all. But in general macros are a place of last resort in C++. To be used if nothing else works. In this case I think a simple static array of tests would have worked just fine. somthing like `typedef pair<string, function<...>>; constexpr static  Test tests[] = {Test("NoLoad", &noLoad), Test("NoStore"), [&](ClosureVersion* cls){Visitor::check(cls, ...))}`